room/compiler-processing/src/main/java/androidx/room/compiler/processing/XProcessingStep.kt

```diff
@@ -85,7 +88,7 @@ internal class JavacProcessingStepDelegate(
         val converted = mutableMapOf<String, Set<XElement>>()
         // create a new x processing environment for each step to ensure it can freely cache
         // whatever it wants and we don't keep elements references across rounds.
-        val xEnv = JavacProcessingEnv(env)
+        val xEnv = JavacProcessingEnv(env, targetLanguage)
```

Comment:
i'm not sure if we can support kotlin here so we might consider eventually throwing Unsupported exception from the constructor of this but fine for this PR until we clarify what to do there.

Reply:
The processor doesn´t work with multiple rounds if it generates kotlin code, but I thought this API was fine.

Reply:
i was more thinking in the lines of "can we fake kotlin structure from java processor". e.g. we would need to detect things like "this class is actually generated from a KSFile" etc, which i thought would be hard.